src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -131,6 +142,42 @@ export class MgtPeoplePicker extends MgtTemplatedComponent {
       `
     );
   }
+  /**
+   * Allows developer to define array of MicrosoftGraph.Person for selectedPeople
+   *
+   * @param {[MicrosoftGraph.UserIdentity]} [id]
+   * @returns {[MicrosoftGraph.Person]}
+   * @memberof MgtPeoplePicker
+   */
+  public async selectUsersById(people) {
+    const provider = Providers.globalProvider;
+    if (provider && provider.state === ProviderState.SignedIn) {
+      const client = Providers.globalProvider.graph;
+      const peopleDetails = [];
+      const name = [];
+      const peopleArr = [];
+      if (people) {
+        // tslint:disable-next-line: prefer-for-of
+        for (let i = 0; i < people.length; i++) {
+          // first find valid UserName
+          peopleDetails.push(await client.getUser(people[i]));
+        }
+        // tslint:disable-next-line: prefer-for-of
+        for (let i = 0; i < peopleDetails.length; i++) {
```

Comment:
no need for 4 for loops, simplify and only use one for loop. You also don't need peopleDetails, name, and peopleArr, arrays

Reply:
Reduced to one loop, I didn't need the extra Graph call actually. 
98eb266